src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_TabKey.cs

```diff
@@ -67,60 +64,91 @@ private bool TryInvokeSnippetCompletion(TabKeyCommandArgs args, CompletionServic
 
             var text = subjectBuffer.AsTextContainer().CurrentText;
 
-            // Delete the ? and invoke completion
+            // If the user types "<line start><spaces><question><tab>"
```

Comment:
> The editor should also shows the full snippet picker UI in the case of <whatever><identifier><spaces><question><tab>

Seems like it.  

> Doesn't it only have to do with what's immediately to the left of the question mark (space or identifier)?

No.  This code here (even before my refactoring) did a lot of extra work to make sure it was at the start of a line (not counting whitespace).  i.f. if you had:

```id id?<tab>```   that would not involve snippets.  THe code in question that prevented this was ```return seenIdentifier && p <= start;```.  Namely, while walking backward, the current position ```p``` had to get to the ```start``` of the current line you were on.